core/client/src/main/java/alluxio/client/block/stream/BlockOutStream.java

```diff
@@ -0,0 +1,181 @@
+ * calling
+ * {@link alluxio.client.block.AlluxioBlockStore#getOutStream(long, long, OutStreamOptions)}.
+ */
+@NotThreadSafe
+public final class BlockOutStream extends FilterOutputStream implements BoundedStream, Cancelable {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private final long mBlockId;
+  private final long mBlockSize;
+  private final Closer mCloser;
+  private final BlockWorkerClient mBlockWorkerClient;
+  private final PacketOutStream mOutStream;
+  private boolean mClosed;
+
+  /**
+   * Creates a new local block output stream.
+   *
+   * @param blockId the block id
+   * @param blockSize the block size
+   * @param workerNetAddress the worker network address
+   * @param context the block store context
+   * @param options the options
+   * @throws IOException if an I/O error occurs
+   * @return the {@link BlockOutStream} instance created
+   */
+  public static BlockOutStream createLocalBlockOutStream(long blockId,
+      long blockSize,
+      WorkerNetAddress workerNetAddress,
+      BlockStoreContext context,
+      OutStreamOptions options) throws IOException {
+    Closer closer = Closer.create();
+    try {
+      BlockWorkerClient client = closer.register(context.createWorkerClient(workerNetAddress));
+      PacketWriter packetWriter =
+          closer.register(LocalFilePacketWriter.createLocalPacketWriter(client, blockId));
+      return new BlockOutStream(blockId, blockSize, client, packetWriter, options);
+    } catch (IOException e) {
+      closer.close();
+      throw e;
+    }
+  }
+
+  /**
+   * Creates a new remote block output stream.
+   *
+   * @param blockId the block id
+   * @param blockSize the block size
+   * @param workerNetAddress the worker network address
+   * @param context the block store context
+   * @param options the options
+   * @throws IOException if an I/O error occurs
+   * @return the {@link BlockOutStream} instance created
+   */
+  public static BlockOutStream createRemoteBlockOutStream(long blockId,
+      long blockSize,
+      WorkerNetAddress workerNetAddress,
+      BlockStoreContext context,
+      OutStreamOptions options) throws IOException {
+    Closer closer = Closer.create();
+    try {
+      BlockWorkerClient client = closer.register(context.createWorkerClient(workerNetAddress));
+      PacketWriter packetWriter = closer.register(
+          new NettyPacketWriter(client.getDataServerAddress(), blockId, blockSize,
+              client.getSessionId(), Protocol.RequestType.ALLUXIO_BLOCK));
+      return new BlockOutStream(blockId, blockSize, client, packetWriter, options);
+    } catch (IOException e) {
+      closer.close();
+      throw e;
+    }
+  }
+
+  @Override
+  public void write(byte[] b) throws IOException {
+    mOutStream.write(b);
+  }
+
+  @Override
+  public void write(byte[] b, int off, int len) throws IOException {
```

Comment:
can you document this to prevent some one accidentally remove your implementation here